src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -45,7 +45,18 @@ void GcsActorScheduler::Schedule(std::shared_ptr<GcsActor> actor) {
   RAY_CHECK(actor->GetNodeID().IsNil() && actor->GetWorkerID().IsNil());
 
   // Select a node to lease worker for the actor.
-  auto node = SelectNodeRandomly();
+  std::shared_ptr<rpc::GcsNodeInfo> node;
+
+  // If an actor is non-detached and has resource requirements, We will try to schedule it
+  // on the same node as the owner if possible.
+  const auto &task_spec = actor->GetCreationTaskSpecification();
+  if (!actor->IsDetached() && !task_spec.GetRequiredResources().IsEmpty()) {
```

Comment:
```suggestion
  if (!task_spec.GetRequiredResources().IsEmpty()) {
```

I think lifetime should be orthogonal.